File: software/examples/c/example_callback.c

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_line.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

// Callback function for reflectivity 
void cb_reflectivity(uint16_t reflectivity, void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("Reflectivity: %d\n", reflectivity);
}

int main() {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	Line line;
	line_create(&line, UID, &ipcon); 

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		exit(1);
	}
	// Don't use device before ipcon is connected

	// Set Period for reflectivity callback to 1s (1000ms)
	// Note: The reflectivity callback is only called every second if the 
	//       reflectivity has changed since the last call!
	line_set_reflectivity_callback_period(&line, 1000);

	// Register reflectivity callback to function cb_reflectivity
	line_register_callback(&line,
	                       LINE_CALLBACK_REFLECTIVITY,
	                       (void *)cb_reflectivity,
	                       NULL);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
}

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletLine.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletLine;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Line Bricklet

// Callback function for reflectivity callback
function cb_reflectivity($reflectivity)
{
    echo "Reflectivity: $reflectivity\n";
}

$ipcon = new IPConnection(); // Create IP connection
$l = new BrickletLine(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register reflectivity callback to function cb_reflectivity
$l->registerCallback(BrickletLine::CALLBACK_REFLECTIVITY, 'cb_reflectivity');

// Set period for reflectivity callback to 1s (1000ms)
// Note: The reflectivity callback is only called every second
//       if the reflectivity has changed since the last call!
$l->setReflectivityCallbackPeriod(1000);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
